presto-main/src/main/java/com/facebook/presto/operator/WorkProcessor.java

```diff
@@ -193,10 +193,6 @@
     @Immutable
     final class ProcessorState<T>
     {
-        private static final ProcessorState NEEDS_MORE_DATE_STATE = new ProcessorState<>(NEEDS_MORE_DATA, true, Optional.empty(), Optional.empty());
```

Comment:
Although probably it doesn't matter, you could run `PagesSortBenchmark#runPagesMergeSortBenchmark` benchmark

Reply:
they showed some small regression. Therefore, i retained the constsnts

Reply:
interesting. Was this within measurement error?

Reply:
I thought there would be no difference. It only matters for distributed sort now anyway.

Another thing is yielding in `MergeSourcePages`. It would be great if it was adaptive on `DriverYieldContext` (https://github.com/prestodb/presto/pull/11816#issuecomment-434625754)